tests/infrastructure/instance_types/test_common_vm_preference.py

```diff
@@ -71,14 +83,15 @@ def start_vm_with_cluster_preference(client, preference_name, namespace_name):
         # TODO add corresponding images to the VM based on preference
         "image": Images.Rhel.RHEL9_REGISTRY_GUEST_IMG,
         "memory_guest": cluster_preference.instance.spec.requirements.memory.guest,
-        "cpu_sockets": cluster_preference.instance.spec.requirements.cpu.guest,
+        "cpu_sockets": sockets,
         "vm_preference": cluster_preference,
     }
 
-    if preference_name == "rhel.9.realtime":
+    # Add cores and threads only if spreadOptions exists
+    if cores is not None and threads is not None:
         vm_args.update({
-            "cpu_cores": cluster_preference.instance.spec.requirements.cpu.guest,
-            "cpu_threads": cluster_preference.instance.spec.requirements.cpu.guest,
+            "cpu_cores": cores,
+            "cpu_threads": threads,
```

Comment:
this `if` not needed, since you assigned None to these variabled - you can add them to `vm_args` dict:
```
            "cpu_cores": cores,
            "cpu_threads": threads,
```